test/units/modules/network/nxos/test_nxos_vlans.py

```diff
@@ -0,0 +1,138 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from textwrap import dedent
+from units.compat.mock import patch
+from units.modules.utils import AnsibleFailJson
+from ansible.modules.network.nxos import nxos_vlans
+from ansible.module_utils.network.nxos.config.vlans.vlans import Vlans
+from .nxos_module import TestNxosModule, load_fixture, set_module_args
+
+ignore_provider_arg = True
+
+
+class TestNxosVlansModule(TestNxosModule):
+
+    module = nxos_vlans
+
+    def setUp(self):
+        super(TestNxosVlansModule, self).setUp()
+
+        self.mock_FACT_LEGACY_SUBSETS = patch('ansible.module_utils.network.nxos.facts.facts.FACT_LEGACY_SUBSETS')
+        self.FACT_LEGACY_SUBSETS = self.mock_FACT_LEGACY_SUBSETS.start()
+
+        self.mock_get_resource_connection_config = patch('ansible.module_utils.network.common.cfg.base.get_resource_connection')
+        self.get_resource_connection_config = self.mock_get_resource_connection_config.start()
+
+        self.mock_get_resource_connection_facts = patch('ansible.module_utils.network.common.facts.facts.get_resource_connection')
+        self.get_resource_connection_facts = self.mock_get_resource_connection_facts.start()
+
+        self.mock_edit_config = patch('ansible.module_utils.network.nxos.config.vlans.vlans.Vlans.edit_config')
+        self.edit_config = self.mock_edit_config.start()
+
+        self.mock_get_device_data = patch('ansible.module_utils.network.nxos.facts.vlans.vlans.VlansFacts.get_device_data')
+        self.get_device_data = self.mock_get_device_data.start()
+
+    def tearDown(self):
+        super(TestNxosVlansModule, self).tearDown()
+        self.mock_FACT_LEGACY_SUBSETS.stop()
+        self.mock_get_resource_connection_config.stop()
+        self.mock_get_resource_connection_facts.stop()
+        self.mock_edit_config.stop()
+
+    def load_fixtures(self, commands=None, device=''):
+        self.mock_FACT_LEGACY_SUBSETS.return_value = dict()
+        self.edit_config.return_value = None
+
+        def load_from_file(*args, **kwargs):
+            cmd = args[1]
+            filename = str(cmd).split(' | ')[0].replace(' ', '_')
+            return load_fixture('nxos_vlans', filename)
+        self.get_device_data.side_effect = load_from_file
+
+    def test_1(self):
+        '''
+        vlan 1,3-5
+        vlan 3
+          name test-vlan3
+        !Note:vlan 4 is present with default settings
+        vlan 5
+          shutdown
+          name test-changeme
+          mode fabricpath
+          state suspend
+          vn-segment 942
+        !Note:vlan 7 is not present
+        vlan 8
+          shutdown
+        '''
+        playbook = dict(config=[
+            dict(vlan_id=4),
+            dict(vlan_id=5, mapped_vni=555, mode='ce'),
```

Comment:
Can you add `name='test-changeme'` to this test for `vlan 5`?  Tests that 3 of the properties are updated and one is not.

Reply:
I added a name attr to vlan 8 instead. 
I'm using vlan 5 name attr to test ignoring existing attrs with merge, as well as cleaning up unspecified attrs with replaced/overridden.